src/main/java/com/salesforce/mirus/HerderStatusMonitor.java

```diff
@@ -91,42 +90,56 @@ private void onConnectors(Throwable error, Collection<String> connectorNames) {
 
   private void processConnector(String connectorName) {
     ConnectorStateInfo stateInfo = herder.connectorStatus(connectorName);
-    if (workerId.equals(stateInfo.connector().workerId())) {
-      if (stateInfo.connector().state().equals(ConnectorStatus.State.FAILED.toString())) {
-        connectorJmxReport.incrementConnectorRestartAttempts(connectorName);
-        if (autoRestartConnectorEnabled) {
-          logger.info("Attempting to restart connector {}", connectorName);
-          herder.restartConnector(
-              connectorName,
-              (error, _void) -> {
-                if (error != null) {
-                  logger.warn("Failed to restart connector {}", connectorName, error);
-                }
-              });
-        }
-      } else {
-        herder.connectorInfo(connectorName, this::onConnectInfo);
-      }
-    }
-  }
-
-  private void onConnectInfo(Throwable error, ConnectorInfo connectorInfo) {
-    if (error != null) {
-      logger.warn("Failed to retrieve connector info, Error details: {}", error);
-      return;
+    boolean isAssignedWorker = workerId.equals(stateInfo.connector().workerId());
+    ConnectorStatus.State connectorState =
+        ConnectorStatus.State.valueOf(stateInfo.connector().state());
+
+    herder.connectorInfo(
+        connectorName,
+        (error, connectorInfo) -> {
+          if (error != null) {
+            logger.warn("Failed to retrieve connector info, Error details: {}", error);
+            return;
+          }
+          // Only the worker assigned to this controller should report connector metrics
+          // Report metrics regardless of current state
+          if (isAssignedWorker) {
+            connectorJmxReport.handleConnector(herder, connectorInfo);
```

Comment:
@dalassi1 Just a note: I think this function doesn't really need to use `connectorInfo`, it actually just needs `stateInfo` so could be pulled right out of the connectorInfo lambda.  I don't want to make too many changes at once though so leaving for now.